### PR TITLE
Add frame skip & frame count functions

### PR DESCRIPTION
**Pull request by ianshmean**
The objective here is to provide the fastest way to 
1) skip through a frame 
2) skip through multiple frames 
3) get the most accurate & fastest count of remaining available frames

@rdeits @yakir12 What do you think about these?

I'll add tests if they look good

### CODE REVIEW

File: src/avio.jl [hunk 1]
```diff
@@ -601,6 +601,49 @@ end
...
+        skipframe(s, throwEOF=throwEOF) && return n
+    end
+end
+
+"""
+    countframes(s::VideoReader)
+    
+Count the number of frames remaining in the video by skipping through each frame.
+"""
+function countframes(s::VideoReader)
```
Comment (rdeits) [inline src/avio.jl:640]: Won't this result in the video being advanced to the end? That is, if I do `countframes(video)` and then try to `read(video)`, won't I get an EOF error?

Comment (ianshmean) [reply src/avio.jl:640]: Indeed. The user would want to `seekstart` afterwards. Or maybe that could be part of the function

Comment (ianshmean) [reply src/avio.jl]: I've changed this to `counttotalframes` which seeks to the start, counts, then seeks back to the start

**Comment by ianshmean** [comment]
`counttotalframes` is currently under-counting by 3 (tested for n=5 and n=100)

```
using VideoIO
n = 100
imgstack = map(x->rand(UInt8,100,100),1:n)
props = [:priv_data => ("crf"=>"22","preset"=>"medium")]
encodedvideopath = VideoIO.encodevideo("testvideo.mp4",imgstack,framerate=30,AVCodecContextProperties=props, silent=true)

f = VideoIO.openvideo(encodedvideopath)
@show counttotalframes(f)
counttotalframes(f) = 97
```

**Comment by yakir12** [comment]
Did I have a sea of similar problems with uncounted frames before... oy...

**Comment by ianshmean** [comment]
It's weird because in the precision tests, we're producing a frame stack, encoding it to file, decoding it and checking every frame against the source framestack.. and those tests pass.

I could understand if it was n-1, perhaps due to the way that EOF is reported, but n-3 seems odd

**Comment by ianshmean** [comment]
Ok, this is related to crf & keyframes. It works for lossless (but it's overcounting by one. I need to fix the EOF  handling):

```
using VideoIO

function testcount(n, crf)
        imgstack = map(x->rand(UInt8,100,100),1:n)
        props = [:priv_data => ("crf"=>crf,"preset"=>"medium")]
        encodedvideopath = VideoIO.encodevideo("testvideo.mp4",imgstack,framerate=30,AVCodecContextProperties=props, silent=true)

        f = VideoIO.openvideo(encodedvideopath)
        return counttotalframes(f)
end

testcount(5, "22")   # 2
testcount(5, "0")    # 6
testcount(5, "1")    # 2
```

**Comment by ianshmean** [comment]
So it seems if `crf > 0` during encoding the total number of frames available to read back in is `n-4`.

Seems like an upstream issue?

**Comment by ianshmean** [comment]
I'd be happy to merge this, if all agree

**Comment by yakir12** [comment]
Sorry for the lack of input, looks good to me!

**Comment by rdeits** [comment]
:+1:

**Pull request merged**